xds/internal/resolver/xds_resolver_test.go

```diff
@@ -390,8 +390,8 @@ func (s) TestXDSResolverResourceNameToWatch(t *testing.T) {
 // from the underlying xdsClient is received after the resolver is closed.
 func (s) TestXDSResolverWatchCallbackAfterClose(t *testing.T) {
 	xdsR, xdsC, tcc, cancel := testSetup(t, setupOpts{target: target})
+	defer xdsR.Close()
```

Comment:
This isn't necessary. Close is already called by `cancel()`.

Reply:
Sorry this is the wrong function, it should be TestXDSResolverWatchCallbackAfterClose. But anyway even in that function, testSetup has already fixed the issue, thank you!